tests/results/DiffTest.py

```diff
@@ -207,6 +209,183 @@ def test_from_string_arrays(self):
+                '+B',
+                '@@ -4,2 +5,3 @@',
+                ' E',
+                '+F',
+                ' G']
+        diff_string = '\n'.join(diff)
+        self.uut = Diff.from_unified_diff(diff_string, source)
+        self.assertEqual(self.uut.original, source)
+        self.assertEqual(self.uut.modified, target)
+
+        source = ['A', 'B', 'C', 'D', 'E', 'F', 'G']
+        target = ['A', 'B', 'Z', 'D', 'E', 'F', 'K']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,3 +1,3 @@',
+                ' A',
+                ' B',
+                '-C',
+                '+Z',
+                '@@ -6,2 +5,2 @@',
+                ' F',
+                '-G',
+                '+K']
+        diff_string = '\n'.join(diff)
+        self.uut = Diff.from_unified_diff(diff_string, source)
+        self.assertEqual(self.uut.original, source)
+        self.assertEqual(self.uut.modified, target)
+
+        # test invalid unified diffs
+        source = ['first', 'second']
+        target = ['only_first_changed', 'second']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,2 +1,2 @@',
+                '-line_to_be_deleted_is_not_same',
+                '+only_first_changed',
+                ' second']
+        diff_string = '\n'.join(diff)
+        with self.assertRaises(RuntimeError):
+            self.uut = Diff.from_unified_diff(diff_string, source)
+
+        source = ['first', 'second']
+        target = ['first', 'second']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,2 +1,2 @@',
+                ' context_line_is_not_same',
+                ' second']
+        diff_string = '\n'.join(diff)
+        with self.assertRaises(RuntimeError):
+            self.uut = Diff.from_unified_diff(diff_string, source)
+
+        # test invalid line type character
+        source = ['first', 'invalid starting character']
+        target = ['first', 'invalid starting character']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,2 +1,2 @@',
+                ' first',
+                '*invalid_starting_character']
+        diff_string = '\n'.join(diff)
+        with self.assertRaises(UnidiffParseError):
+            self.uut = Diff.from_unified_diff(diff_string, source)
+
+        # test invalid hunks
+        source = ['A', 'B', 'C', 'D', 'E', 'F', 'G']
+        target = ['A', 'B', 'Z', 'D', 'E', 'F', 'K']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,7 +1,5 @@',
+                ' A',
+                ' B',
+                '-C',
+                '+Z',
+                '@@ -6,2 +5,2 @@',
+                ' F',
+                '-G',
+                '+K']
+        diff_string = '\n'.join(diff)
+        with self.assertRaises(UnidiffParseError):
+            self.uut = Diff.from_unified_diff(diff_string, source)
```

Comment:
could you split up each test setup into different test functions? I know finding names for them is hard, though try to name the aspect you try to test inside the test-function, like
`test_from_unified_diff_invalid_hunk`, `test_from_unified_diff_simple`, `test_from_unified_diff_2_changes_beneath` etc etc